source/gwork/include/Gwork/Skins/TexturedBase.h

```diff
@@ -1006,7 +1003,7 @@ namespace Gwk
                 Textures.Input.Slider.H.Normal.DrawCenter( GetRender(), control->GetRenderBounds() );
             }
 
-            void DrawTreeNode( Controls::Base* ctrl, bool bOpen, bool bSelected, int iLabelHeight, int iLabelWidth, int iHalfWay, int iLastBranch, bool bIsRoot )
+            virtual void DrawTreeNode( Controls::Base* ctrl, bool bOpen, bool bSelected, int iLabelHeight, int iLabelWidth, int iHalfWay, int iLastBranch, bool bIsRoot ) override
```

Comment:
The override is the important bit. The virtual here is redundant, and if anything misleading. If you only see virtual at the first parent method then you know that is the "base". When you see override you know a) it is overriding something and b) the compiler will check that the override is correct.

Override is great in a codebase like this with loads of child classes when you are refactoring heavily. It is really annoying tracking down bugs due to virtual failures. :(

If you don't mind removing the redundant virtuals (i.e. where it is an override) that would be great.